src/WrappedMessage.js

```diff
@@ -0,0 +1,45 @@
+
+const messagesInFlight = new Set();
+
+export class WrappedMessage {
+  constructor(message) {
+    const { nack, reject, ack, rabbotMessage, ...rest } = message;
+    Object.assign(this, rest);
+    this.rabbotMessage = message;
+    messagesInFlight.add(this);
+  }
+
+  async ack() {
+    try {
+      this.rabbotMessage.ack();
```

Comment:
No await? Did we decide that was superstition?

Reply:
No, you were right, I forgot that. Added it back in.